src/msg/async/dpdk/DPDK.h

```diff
@@ -769,7 +774,33 @@ class DPDKDevice {
    */
   int init_port_fini();
 
+  void nic_stats_dump(bufferlist &bl);
+  void nic_xstats_dump(bufferlist &bl);
  private:
+  class XstatSocketHook:public AdminSocketHook {
```

Comment:
might want to define this derived class in the .cc file. and keep an instance of `std::unique_ptr<AdminSocketHook>` as a member variable instead.